Modules/TPC/src/PID.cxx

```diff
@@ -97,6 +101,48 @@ void PID::monitorData(o2::framework::ProcessingContext& ctx)
 
 void PID::endOfCycle()
 {
+  // ===| Fitting Histogram for separation Power |============================================================
+  std::unique_ptr<TF1> fitFunc = std::make_unique<TF1>("fitFunc", "[0]*exp(-0.5*((x-[1])/[2])^2) + [3]*exp(-0.5*((x-[4])/[5])^2)", 0, 100);
+
+  const int nPars = 8; //6 fit parameters + 2 external (seperation power + chi²/ndf)
+
+  for (auto const& pair : mQCPID.getMapOfHisto()) {
+    for (auto& hist : pair.second) {
+      if (pair.first.compare("hdEdxMaxMIP") == 0) {
+        mTrendingParameters.clear();
+        // Define fitting function: sum of two Gaussians with an offset
+        // Set initial parameters for the fit
+        fitFunc->SetParameter(0, 3000); // Amplitude of the first Gaussian
+        fitFunc->SetParameter(1, 50);   // Mean of the first Gaussian
+        fitFunc->SetParLimits(1, 45, 55);
+        fitFunc->SetParameter(2, 2);   // Sigma of the first Gaussian
+        fitFunc->SetParameter(3, 100); // Amplitude of the second Gaussian
+        fitFunc->SetParameter(4, 75);  // Mean of the second Gaussian
+        fitFunc->SetParLimits(4, 60, 90);
+        fitFunc->SetParameter(5, 10); // Sigma of the second Gaussian
+
+        // Fit the histogram with the fitting function
+        hist->Fit(fitFunc.get(), "QRN");
+        for (int iPar = 0; iPar < nPars - 2; iPar++) {
+          mTrendingParameters.emplace_back(fitFunc->GetParameter(iPar)); //save fit parameters
+        }
+        // Retrieve parameters of the fitted function
+
+        const double sepPow = (mTrendingParameters[4] - mTrendingParameters[1]) / (mTrendingParameters[2] / 2. + mTrendingParameters[5] / 2.); //separation power
+        mTrendingParameters.emplace_back(sepPow);
+        mTrendingParameters.emplace_back(fitFunc->GetChisquare() / fitFunc->GetNDF()); //chi²/ndf of fit
+      }
+    }
+  }
+
+  pSeparationPower = new TProfile("pSeparationPower", "pSeparationPower", nPars, 0., (float)nPars);
+  getObjectsManager()->startPublishing(pSeparationPower);
+  const TString binLabels[nPars] = { "Amplitude Pi", "Mean Pi", "Sigma Pi", "Amplitude El", "Mean El", "Sigma El", "Separation Power", "chiSquare/ndf" };
```

Comment:
Is it foreseen to create and start publishing in `endOfCycle`. Like this, a new TProfile is created in each cycle which is a potential memory leak. Why is it not created in `initialize`?